Проектная работа Спринт 9

KateSilencio opened the pull request.

KateSilencio committed:
Проектная работа Спринт 9

faritowich commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -2,10 +2,65 @@ package com.example.playlistmaker
+import android.widget.ImageView
 
 class SearchActivity : AppCompatActivity() {
+    var edit = ""
```
private

faritowich commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -2,10 +2,65 @@ package com.example.playlistmaker
+                if (s.isNullOrEmpty()) {
+                    clearButton.visibility = View.GONE
+                } else {
+                    clearButton.visibility = View.VISIBLE
```
Для изменения видимости элемента на экране в данном случае удобнее использовать extension View.isVisible:
View.isVisible = true соответствует View.visibility = View.VISIBLE
View.isVisible = false соответствует View.visibility = View.GONE

faritowich commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -2,10 +2,65 @@ package com.example.playlistmaker
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putString("EditText",edit)
```
"EditText" и вообще строки-ключи стоит выносить в отдельную константу (в данном случае в companion object). Это позволит не бояться забыть поменять в одном месте и забыть в другом. При необходимости просто меняешь в самой константе и всё так же работает. Плюс сразу понятно, где мы передаем и получаем эти данные.

faritowich commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -1,18 +1,54 @@
+                putExtra(Intent.EXTRA_SUBJECT,getString(R.string.topic_mail))
+                putExtra(Intent.EXTRA_TEXT,getString(R.string.message_mail))
+            }
+            startActivity(mailIntent)
```
Можно упростить до
```
Intent(Intent.ACTION_SENDTO).apply {
    data = Uri.parse("mailto:")
    putExtra(Intent.EXTRA_EMAIL, arrayOf(getString(R.string.student_mail)))
    putExtra(Intent.EXTRA_SUBJECT,getString(R.string.topic_mail))
    putExtra(Intent.EXTRA_TEXT,getString(R.string.message_mail))
    startActivity(this)
}
```

faritowich commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -1,9 +1,48 @@
+            style="@style/TitleText"
+            android:text="@string/search" />
+
+    </LinearLayout>
```
Для тулбара можно также использовать androidx.appcompat.widget.Toolbar

faritowich commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -1,9 +1,48 @@
+            android:src="@drawable/ic_clear"
+            android:visibility="gone" />
+
+    </RelativeLayout>
```
RelativeLayout лучше не использовать - этот лейаут уже больше относится к пережитку прошлого, хоть и до сих пор встречается в старых проектах и полезно знать, как он работает. А если верстать по новой, лучше использовать ConstraintLayout - он намного эффективнее.

Но тут ни один из них и не нужен, тут нужен простой лейаут типа FrameLayout

faritowich commented on app/src/main/res/values-night/styles.xml:
```diff
@@ -0,0 +1,74 @@
+    <style name="ButtonStyleMainScr" parent="Widget.MaterialComponents.Button.UnelevatedButton">
+        <item name="android:layout_width">match_parent</item>
+        <item name="android:layout_height">0dp</item>
+        <item name="android:layout_marginHorizontal">16dp</item>
```
Лучше внешние отступы, высоту и ширину, веса, гравити не выносить в стиль. Если на экране будет много элементов, будет сложнее ориентироваться с их взаимным расположением, когда отвечающие за это атрибуты разнесены по разным файлам - что-то может быть в лейауте, что-то в стиле. В стилях обычно содержат атрибуты, отвечающие за внешний вид вьюхи и вообще только за то, что находится внутри нее - цвета, шрифты, фоны и т.д. Бывают исключения, когда, например, мы точно знаем, что у кнопки всегда должна строго определенная высота и ширина - тогда да, имеет смысл выносить

KateSilencio committed:
Проектная работа Спринт 9

Исправленная версия

KateSilencio committed:
Проектная работа Спринт 10
 Работа со списками

KateSilencio reviewed (commented):
Забыла замержить 9 спринт. Сделаю merge всех веток уже после проверки и правки практической 10 спринта

kekulta commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,11 +1,137 @@
+
+        val inputEditText = findViewById<EditText>(R.id.edit_text_search_id)
+        val clearButton = findViewById<ImageView>(R.id.clear_btn_id)
+        //Установить восстановленные данные в EditText
```
Здорово, что пишешь комментарии

kekulta commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,11 +1,137 @@
+
+            override fun afterTextChanged(s: Editable?) {
+                if (s.isNullOrEmpty()) {
+                    //clearButton.visibility = View.GONE
```
А вот закомментированный код лучше никогда не коммитить

kekulta commented on app/src/main/java/com/example/playlistmaker/TracksViewHolder.kt:
```diff
@@ -0,0 +1,41 @@
+            .into(artworkUrl)
+    }
+
+    //конвертация dp в px
+    fun dpToPx(dp: Float, context: Context): Int {
+        return TypedValue.applyDimension(
+            TypedValue.COMPLEX_UNIT_DIP,
+            dp,
+            context.resources.displayMetrics
+        ).toInt()
+    }
```
Эта функция бывает часто нужна, если делаешь вёрстку из кода. Можно написать её как экстеншен функцию над контекстом.

kekulta commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,11 +1,137 @@
+            )
+        )
+
+        private const val EDIT_TEXT = "EditText"
```
Молодец,что клюси выносишь в константу.

KateSilencio committed:
Проектная работа Спринт 10
 Работа со списками
 Версия 2
 Исправлено:
 1. в ic_search изменены размеры
 2. в track_view.xml в ImageView убран padding
 3. в ConstraintLayout изменены paddings
 4. Добавлен файл ContextExtensions.kt с функцией расширением Context.dpToPx

KateSilencio committed:
Проектная работа Спринт 10
 Работа со списками
 Версия 3
 Исправлено:
 1. исправлены paddings в track_view.xml
